### PR TITLE
Use assert module's strict mode

### LINKED ISSUE #121: assert.equal() is deprecated
Per, [Node API docs](https://nodejs.org/api/assert.html#assert_assert_equal_actual_expected_message) `assert.equal()` is deprecated since v9.9.0 &mdash; use `strictEqual()` instead.

```js
/** @deprecated since v9.9.0 - use strictEqual() instead. */
function equal(actual: any, expected: any, message?: string | Error): void;
```

Trying to find an ESLint config that can flag deprecated Node APIs for us. Early research shows https://github.com/mysticatea/eslint-plugin-node should do the trick, but I need to figure out how to configure it and silence some of the noisier recommended rules.

### PR DESCRIPTION
**Pull request by pdehaan**
Fixes #121

### CODE REVIEW

File: tests/core-addon/IonCore.test.js [hunk 1]
```diff
@@ -427,7 +427,7 @@ describe('IonCore', function () {
   describe('_updateInstalledStudies()', function () {
     it('adds the ionInstalled property', async function () {
       // We don't expect any update task to be running now.
-      assert.equal(this.ionCore._updateInstalledTask, null);
+      assert.equal(this.ionCore._updateInstalledTask, undefined);
```
Comment (Dexterp37) [inline tests/core-addon/IonCore.test.js:430]: AH! This is vestigial code, it's testing something that's no longer there :)
Would you kindly remove this line and the comment immediately above?

**Review by Dexterp37 — CHANGES_REQUESTED** [review]
Thanks again for the nice PR!

File: tests/core-addon/IonCore.test.js (hunk 1 above)
Comment (pdehaan) [reply tests/core-addon/IonCore.test.js]: Removed.

**Review by Dexterp37 — APPROVED** [review]

File: tests/core-addon/IonCore.test.js (hunk 1 above)
Comment (Dexterp37) [reply tests/core-addon/IonCore.test.js]: Awesome, thank you so much!

**Pull request merged**